scripts/gen-deps-list.js

```diff
@@ -1,110 +1,101 @@
 #!/usr/bin/env node
 'use strict'
 
-const { program, Argument } = require('commander')
 const DepTree = require('deptree')
 const fs = require('fs').promises
 const joinPath = require('path').join
 const semver = require('semver')
 const { getPackages } = require('./utils')
 const escapeRegExp = require('lodash/escapeRegExp')
 const invert = require('lodash/invert')
+const trim = require('lodash/trim')
 
 const changelogConfig = {
   path: joinPath(__dirname, '../CHANGELOG.unreleased.md'),
   startTag: '<!--packages-start-->',
   endTag: '<!--packages-end-->',
 }
 
-program
-  .argument('<package-name>', 'The name of the package to release')
-  .addArgument(new Argument('<release-type>', 'The type of release to perform').choices(['patch', 'minor', 'major']))
-  .option('-r, --read-changelog', 'Import existing packages from the changelog')
-  .option('-w, --write-changelog', 'Write output to the changelog')
-  .option('--force', 'Required when using --write-changelog without --read-changelog')
-  .showHelpAfterError(true)
-  .showSuggestionAfterError(true)
-  .parse()
-
-const [rootPackageName, rootReleaseType] = program.args
-const { readChangelog, writeChangelog, force } = program.opts()
-
-if (writeChangelog && !readChangelog && !force) {
-  // Stop the process to prevent unwanted changelog overwrite
-  program.showHelpAfterError(false).error(`
-  WARNING: Using --write-changelog without --read-changelog will remove existing packages list.
-  If you are sure you want to do this, add --force.
-  `)
-}
-
 const RELEASE_WEIGHT = { PATCH: 1, MINOR: 2, MAJOR: 3 }
 const RELEASE_TYPE = invert(RELEASE_WEIGHT)
-const rootReleaseWeight = releaseTypeToWeight(rootReleaseType)
 
+const dependencyTree = new DepTree()
 /** @type {Map<string, int>} A mapping of package names to their release weight */
-const packagesToRelease = new Map([[rootPackageName, rootReleaseWeight]])
+const packagesToRelease = new Map()
 
-const dependencyTree = new DepTree()
+let allPackages
 
 async function main() {
-  if (readChangelog) {
-    await importPackagesFromChangelog()
+  allPackages = await getPackages(true)
+  const content = await fs.readFile(changelogConfig.path)
+  const changelogRegex = new RegExp(
+    `${escapeRegExp(changelogConfig.startTag)}(.*)${escapeRegExp(changelogConfig.endTag)}`,
+    's'
+  )
+  const block = changelogRegex.exec(content)?.[1].trim()
+
+  if (block === undefined) {
+    throw new Error(`Could not find changelog block in ${changelogConfig.path}`)
   }
 
-  const packages = await getPackages(true)
-  const rootPackage = packages.find(pkg => pkg.name === rootPackageName)
+  const lines = block.split(/\s*\r?\n\s*/).map(line => {
+    const items = trim(line, '- ').split(/\s+/)
```

Comment:
Can we use a regex here instead?

I believe it makes the format clearer, and we can also handle release type validation at the same time :)

```suggestion
    const matches = /^-\s*(\S+)\s+(patch|minor|major)$/
    if (matches === null) {
      throw new Error(`Invalid line: "${line}"`)
    }
    return matches.slice(1)
```